bundles/binding/org.openhab.binding.temperlan/src/main/java/org/openhab/binding/temperlan/internal/TemperlanGenericBindingProvider.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.temperlan.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openhab.binding.temperlan.TemperlanBindingProvider;
+import org.openhab.binding.temperlan.internal.TemperlanBindingConfig;
+import org.openhab.core.binding.BindingConfig;
+import org.openhab.core.items.Item;
+import org.openhab.model.item.binding.AbstractGenericBindingProvider;
+import org.openhab.model.item.binding.BindingConfigParseException;
+
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * 
+ * @author snoerenberg
+ * @since 1.0
+ */
+public class TemperlanGenericBindingProvider extends AbstractGenericBindingProvider implements TemperlanBindingProvider {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public String getBindingType() {
+		return "temperlan";
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void validateItemType(Item item, String bindingConfig) throws BindingConfigParseException {
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void processBindingConfiguration(String context, Item item, String bindingConfig) throws BindingConfigParseException {
+		super.processBindingConfiguration(context, item, bindingConfig);

```

Comment:
move to end of method in case other code throw exception, thereby leaving bad data in internal structures.